saltboot-formula/_states/saltboot.py

```diff
@@ -1666,18 +1685,25 @@ def verify_and_boot_system(name, partitioning, images, boot_images, action = 'fa
         return ret
 
     if action == 'kexec':
-        # try to load kexec
-        cmd = 'set -e -x ;'
-        cmd += 'mkdir -p /kexec_tmp ; umount /kexec_tmp || true ;'
-        cmd += 'mount -t tmpfs -o size=500M,nr_inodes=1k,mode=0755 tmpfs /kexec_tmp ;'
-        cmd += _download_url_cmd(boot_image.get('kernel', {}).get('url', ''), '/kexec_tmp/kernel') + ' ;'
-        cmd += _download_url_cmd(boot_image.get('initrd', {}).get('url', ''), '/kexec_tmp/initrd') + ' ;'
-        cmd += 'kexec -l /kexec_tmp/kernel --reuse-cmdline --append="$(cat /update_kernel_cmdline)" --initrd=/kexec_tmp/initrd'
-
-        res = __salt__['cmd.run_all'](cmd, python_shell=True)
-        if res['retcode'] > 0:
-            ret['comment'] += 'Kexec failed, doing normal reboot.\n' + res['stdout'] + res['stderr']
+        try:
+            kernel_url = _download_url_cmd(boot_image.get('kernel', {}).get('url', ''), '/kexec_tmp/kernel')
+            initrd_url = _download_url_cmd(boot_image.get('initrd', {}).get('url', ''), '/kexec_tmp/initrd')
```

Comment:
these variables should be renamed, they contain bash command, not url

Reply:
Done